### PR TITLE
[PPML] Refine PPML Readthedoc User Guide

### PR DESCRIPTION
**Pull request by yizerozhuang**

### CODE REVIEW

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ cd BigDL/ppml/
     ./generate-password.sh used_password_when_generate_keys
     cd ..
     ```
-    This scrips will generate 2 files in `./ppml/scripts/password` dir.
+    This scrip will generate 2 files in `./ppml/scripts/password` dir.
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:78]: script

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 2]
```diff
@@ -106,31 +106,31 @@ Enter `BigDL/ppml/trusted-big-data-ml/scala/docker-graphene` dir.
...
-   To train a model with ppml in bigdl, you need to prepare the data first. The Docker image is taking lenet and mnist as example. <br>
+   To train a model with PPML in BigDL, you need to prepare the data first. The Docker image is taking lenet and mnist as examples. <br>
    You can download the MNIST Data from [here](http://yann.lecun.com/exdb/mnist/). Unzip all the files and put them in one folder(e.g. mnist). <br>
    There are four files. **train-images-idx3-ubyte** contains train images, **train-labels-idx1-ubyte** is train label file, **t10k-images-idx3-ubyte** has validation images    and **t10k-labels-idx1-ubyte** contains validation labels. For more detail, please refer to the download page. <br>
    After you decompress the gzip files, these files may be renamed by some decompress tools, e.g. **train-images-idx3-ubyte** is renamed to **train-images.idx3-ubyte**. Please change the name back before you run the example.  <br>
    
-3. To start the container, first modify the paths in deploy-local-spark-sgx.sh, and then run the following commands:
+3. To start the container, firstly modify the paths in deploy-local-spark-sgx.sh, and then run the following commands:
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:119]: Remove firstly may be better

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 3]
```diff
@@ -106,31 +106,31 @@ Enter `BigDL/ppml/trusted-big-data-ml/scala/docker-graphene` dir.
...
+    **DATA_PATH** means the absolute path to the data(like mnist) that would use later in the spark program. According to the above commands, the path would be like "BigDL/ppml/trusted-big-data-ml/scala/docker-graphene/mnist" <br>
     **KEYS_PATH** means the absolute path to the keys you just created and copied to. According to the above commands, the path would be like "BigDL/ppml/trusted-big-data-ml/scala/docker-graphene/keys" <br>
     **LOCAL_IP** means your local IP address. <br>
 
 ##### 2.2.2.2 Run Your Spark Program with BigDL PPML on SGX
 
-To run your pyspark program, first you need to prepare your own pyspark program and put it under the trusted directory in SGX  `/ppml/trusted-big-data-ml/work`. Then run with `ppml-spark-submit.sh` using the command:
+To run your pyspark program, firstly you need to prepare your own pyspark program and put it under the trusted directory in SGX  `/ppml/trusted-big-data-ml/work`. Then run with `ppml-spark-submit.sh` using the command:
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:133]: Remove firstly

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 4]
```diff
@@ -156,17 +156,17 @@ Open another terminal and check the log:
...
+First, download and install sbt from [here](https://www.scala-sbt.org/download.html) and deploy an Hadoop Distributed File System from [here](https://hadoop.apache.org/docs/r2.7.7/hadoop-project-dist/hadoop-common/ClusterSetup.html) for the Transaction Processing Performance Council Benchmark H (TPC-H) dataset and output, then build the source codes with SBT and generate the TPC-H dataset according to the TPC-H example from [here](https://github.com/intel-analytics/zoo-tutorials/tree/master/tpch-spark). After that, check if there is a  `spark-tpc-h-queries_2.11-1.0.jar` under `tpch-spark/target/scala-2.11`; if so, we have successfully packaged the project.
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:167]: HDFS (Hadoop Distributed File System)

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 5]
```diff
@@ -1,63 +1,63 @@
...
+## 1. Introduction
+Protecting privacy and confidentiality is critical for large-scale data analysis and machine learning. BigDL ***PPML*** combines various low-level hardware and software security technologies (e.g., [Intel® Software Guard Extensions (Intel® SGX)](https://www.intel.com/content/www/us/en/architecture-and-technology/software-guard-extensions.html#:~:text=Intel%C2%AE%20SGX%20allows%20user,level%20of%20control%20and%20protection.), [Library Operating System (LibOS)](https://events19.linuxfoundation.org/wp-content/uploads/2017/12/Library-OS-is-the-New-Container-Why-is-Library-OS-A-Better-Option-for-Compatibility-and-Sandboxing-Chia-Che-Tsai-UC-Berkeley.pdf) such as [Graphene](https://www.usenix.org/conference/atc17/technical-sessions/presentation/tsai) and [Occlum](https://github.com/occlum/occlum), Federated Learning, etc.), so that users can continue to apply standard Big Data and AI technologies (such as Apache Spark, Apache Flink, Tensorflow, PyTorch, etc.) without sacrificing privacy.
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:4]: (https://www.intel.com/content/www/us/en/architecture-and-technology/software-guard-extensions.html) is enough. Remove other parameters

Comment (qiyuangong) [reply docs/readthedocs/source/doc/PPML/Overview/ppml.md:4]: [Graphene](https://github.com/gramineproject/graphene)

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 6]
```diff
@@ -1,63 +1,63 @@
...
 
 ## 1.1 PPML for Big Data AI
-BigDL provides a distributed PPML platform for protecting the *end-to-end Big Data AI pipeline* (from data ingestion, data analysis, all the way to machine learning and deep learning). In particular, it extends the single-node [Trusted Execution Environment](https://en.wikipedia.org/wiki/Trusted_execution_environment) to provide a *Trusted Cluster Environment*, so as to run unmodified Big Data analysis and ML/DL programs in a secure fashion on (private or public) cloud:
+BigDL provides a distributed PPML platform for protecting the *end-to-end Big Data AI pipeline* (from data ingestion, data analysis, all the way to machine learning and deep learning). In particular, it extends the single-node [Trusted Execution Environment](https://en.wikipedia.org/wiki/Trusted_execution_environment) to provide a *Trusted Cluster Environment*, so as to run unmodified Big Data analysis and ML/DL programs in a secure fashion on a (private or public) cloud:
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:7]: remove a.

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 7]
```diff
@@ -1,63 +1,63 @@
...
 
  * Compute and memory protected by SGX Enclaves
- * Network communication protected by remote attestation and TLS
+ * Network communication protected by remote attestation and [Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security)
  * Storage (e.g., data and model) protected by encryption
- * Optional federated learning support
+ * Optional Federated Learning support
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:12]: Add link for Federated Learning https://en.wikipedia.org/wiki/Federated_learning

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md [hunk 8]
```diff
@@ -1,63 +1,63 @@
...
+    Check SGX driver with `ls /dev | grep sgx`. If SGX driver is not installed, please install SGX Data Center Attestation Primitives driver from [here](https://github.com/intel/SGXDataCenterAttestationPrimitives/tree/master/driver/linux):
 
     ```bash
     cd scripts/
     ./install-graphene-driver.sh
     cd ..
     ```
 
-2. Generate key for SGX enclave
+2. Generate the key for SGX Enclaves
```
Comment (qiyuangong) [inline docs/readthedocs/source/doc/PPML/Overview/ppml.md:43]: Change the key to signing key

File: docs/readthedocs/source/doc/PPML/Overview/ppml.md (hunk 4 above)
Comment (qiyuangong) [reply docs/readthedocs/source/doc/PPML/Overview/ppml.md:167]: check if there is a . remove a

**Review by qiyuangong — APPROVED** [review]
LGTM
